### PR TITLE
Added option for admin to select default state/province

### LINKED ISSUE #2036: Add setting to select default state
## Issue Overview

Right now we can set the default country. But we should also allow admins to select the default _state_ if applicable.

For instance, if I run a nonprofit that's specific to California I would want the state to always be CA by default. This would allow easier donations for state residents because they wouldn't have to select their state. 

## Expected Behavior
Admins can select the default state in general settings.

## Current Behavior
There is no option to select the default state.

## Possible Solution
Create a new field below "Base country" that is "Default state/province" and have this field conditional based on the base country. If you change from the US for instance to another country with states/provides/whatever then it will update accordingly.

### PR DESCRIPTION
**Pull request by Sidsector9**
## Description
Admin can select default state/province (if any) while selecting/changing the country.
This PR will resolve #2036

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

Related #2036

### CODE REVIEW

File: includes/admin/class-give-settings.php [hunk 1]
```diff
@@ -245,6 +245,13 @@ public function give_settings( $active_tab ) {
 							'options' => give_get_country_list(),
 						),
 						array(
+							'name'    => esc_html__( 'Default State/Province', 'give' ),
```
Comment (ravinderk) [inline includes/admin/class-give-settings.php]: @Sidsector9 There is no need to add setting here. Please remove it.

File: includes/admin/settings/class-settings-general.php [hunk 2]
```diff
@@ -228,13 +228,20 @@ public function get_settings() {
...
-                            'url'   => esc_url( 'http://docs.givewp.com/settings-general' ),
-                            'title' => __( 'General Options', 'give' ),
-                            'type'  => 'give_docs_link',
-                        ),
+						array(
+							'name'    => esc_html__( 'Default State/Province', 'give' ),
+							'desc'    => esc_html__( 'The state/province your site operates from.', 'give' ),
+							'id'      => 'base_state',
+							'type'    => 'select',
+							'options' => give_states_list()[ get_option( 'give_settings' )['base_country'] ],
```
Comment (ravinderk) [inline includes/admin/settings/class-settings-general.php]: @Sidsector9 Give support PHP 5.3+. Did you test with minimum supported version (array short tag will not work in that version)?

**Comment by ravinderk** [comment]
@shivapoudel Thank you for PR. 

Please check my review on this PR.

File: includes/admin/settings/class-settings-general.php (hunk 2 above)
Comment (Sidsector9) [reply includes/admin/settings/class-settings-general.php]: @ravinderk I slept before checking the Travis build completed :sweat_smile:

**Comment by ravinderk** [comment]
@Sidsector9 Did you use Give donation form before developing this? because On Donation form state already update via ajax when contry change.

In the current scenario, when you change country state list will update automatically without set any default state. So we are trying you improve UX here.

In this issue, we want to create a setting as you did in this PR to set default state but you did not use it any where, So use already create Ajax to set default state.

Set default state in `give_ajax_get_states_field` function
https://github.com/WordImpress/Give/blob/master/includes/ajax-functions.php#L194

For Ref:
Frontend: https://github.com/WordImpress/Give/blob/master/assets/js/admin/admin-scripts.js#L1046 
Backend: https://github.com/WordImpress/Give/blob/master/assets/js/admin/admin-scripts.js#L277

Let me know if you have any question.

**Comment by ravinderk** [comment]
@Sidsector9 Closing this PR because @raftaar1191  taking over this.

Thank you 👍

**Comment by Sidsector9** [comment]
Ah I see what you mean! I should have used the built-in function instead of writing a custom one!
To answer your first question, I have only used Give on a testing level.

Thank you for providing the links, that will help for future contributions! :smile:

**Comment by ravinderk** [comment]
@Sidsector9 You can join our community slack channel.  Feel free to ask me development related queries. https://givewp.com/join-us-slack/

**Pull request closed without merging**